lib/ansible/plugins/filter/aws_kms.py

```diff
@@ -0,0 +1,155 @@
+"""
+(c) 2018, Archie Gunasekara <contact@achinthagunasekara.com>
+Module to handle encrypting and decrypting of items with KMS
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+"""
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import base64
+from os.path import basename
+from ansible.errors import AnsibleFilterError
+
+try:
+    import boto3
+    from aws_encryption_sdk.key_providers.kms import KMSMasterKey
+    from aws_encryption_sdk import KMSMasterKeyProvider, encrypt, decrypt
+    from aws_encryption_sdk.exceptions import AWSEncryptionSDKClientError
+    HAS_DEPENDENCIES = True
+except ImportError:
+    HAS_DEPENDENCIES = False
+
+
+def role_arn_to_session(role_arn, role_session_name):
+    """
+    Get credentials to using a given AES role.
+    Args:
+        role_arn: ARN of the role to use to get credentials.
+        role_session_name: A session name to use.
+    Returns:
+        boto3.Session: Returns a boto3.Session object.
+    """
+    client = boto3.client('sts')
+    response = client.assume_role(
+        RoleArn=role_arn,
+        RoleSessionName=role_session_name
+    )
+    return boto3.Session(
+        aws_access_key_id=response['Credentials']['AccessKeyId'],
+        aws_secret_access_key=response['Credentials']['SecretAccessKey'],
+        aws_session_token=response['Credentials']['SessionToken'])
+
+
+def get_boto_session(**kwargs):
+    """
+    Get boto3 session object.
+    Returns:
+        boto3.Session: Returns a boto3.Session object.
+    """
+    valid_keys = [
+        'profile_name',
```

Comment:
I'm a little worried about the discrepancy between the keys here and what AWS modules and plugins accept. This should probably be documented as well, so the user doesn't have to read the code to figure out what is expected of them.

Reply:
Perhaps just keep the keys in sync with other stuff, no?

Reply:
@s-hertel  @defionscode 

I've made these changes. Could you please have a look?